### ISSUE TITLE
Dropping files (`ondrop`) doesn't work on Windows

### ISSUE THREAD

**Issue by Boscop** [DioxusLabs/dioxus#2954]
**Problem**

Dropping files (`ondrop`) doesn't work on Windows, the even doesn't seem to get triggered.

**Steps To Reproduce**

- `cargo r --example file_upload`
- Click on "Choose Files" (the file input) and select a .txt file -> That works fine but:
- Now click "Clear Files" and drag the same .txt file from Windows Explorer and drop it onto the "Drop files here" area
- Nothing happens!

**Expected behavior**

It should load the file just like when opening it by clicking on the file input and selecting it through the file dialog.

**Screenshots**

![image](https://github.com/user-attachments/assets/a6dea390-58c4-482b-bd06-31362af16be0)

**Environment:**
 - Dioxus version:  latest `main` branch, commit `b20db13e84227b76741e007b2763adc8237ada48`
 - Rust version:    `rustc 1.82.0-nightly (8b3870784 2024-08-07)`
 - OS info:         Windows 11
 - App platform:    desktop

**Comment by matthunz** [comment]
Thanks for reporting this!

Looks like it is working on MacOS (and no open issues I can see for `tao`) 
<img width="791" alt="Screenshot 2024-09-12 at 6 44 19 PM" src="https://github.com/user-attachments/assets/3812bfca-7360-4049-8335-d5c7ffcec21d">

**Comment by Boscop** [comment]
@matthunz Yeah, IIRC I reported an issue with file dropping on Windows when the Tauri project was in early stages, not sure on which repo though, I couldn't find the issue or if it was solved..

**Comment by DogeDark** [comment]
I can verify that this does not work on Windows 10, Dioxus main.

**Comment by DogeDark** [comment]
@Boscop I'm unsure. I only tested to verify the issue.

**Comment by Boscop** [comment]
Apparently it's not a tao/wry issue: https://github.com/tauri-apps/tao/issues/986#issuecomment-2392671741
> The tao drag and drop events work just fine on Windows, you can run cargo run --example window and drag a file onto the window, and the event will be logged.
I think dioxus is using wry as well, and in that case you might be looking for [docs.rs/wry/latest/wry/struct.WebViewBuilder.html#method.with_drag_drop_handler](https://docs.rs/wry/latest/wry/struct.WebViewBuilder.html#method.with_drag_drop_handler)

So it must be a dioxus issue then..

**Comment by DogeDark** [comment]
I did some digging and it seems like `with_drag_drop_handler` blocks the HTML `ondrag`, `ondrop` and `ondragover` events on Windows. If we remove it, our event handlers get called correctly but we rely on `with_drag_drop_handler` for the file so the app panics. It looks like `with_drag_drop_handler` has different behavior depending on the platform (or system webview?) since it worked on MacOS.

**Comment by Boscop** [comment]
@DogeDark Hm, what do you mean by "If we remove it, our event handlers get called correctly but we rely on `with_drag_drop_handler` for the file so the app panics."?

Shouldn't the HTML drag/drop events be working in conjunction with / backed by `with_drag_drop_handler`?

**Comment by DogeDark** [comment]
@Boscop We have a JavaScript shim that is called by Dioxus to create event listeners and when `with_drag_drop_handler` is used, the JS-native event listeners I mentioned aren't called at all. The `with_drag_drop_handler` only sets the latest file(s) that were hovered over the application. Our Rust code that handles the file drop event expects the `with_drag_drop_handler` to have set a value but it hasn't so it panics.

**Comment by Boscop** [comment]
@DogeDark
> The with_drag_drop_handler only sets the latest file(s) that were hovered over the application. 

I found that here
https://github.com/DioxusLabs/dioxus/blob/9ffd4b816b8ce291b6ca66cb5c871e107a964f6b/packages/desktop/src/webview.rs#L233

And about this:
> Our Rust code that handles the file drop event expects the with_drag_drop_handler to have set a value but it hasn't

Why doesn't it set the value? :)

**Comment by DogeDark** [comment]
@Boscop It does, it's just with the drop handler, the HTML events don't get called. Without the drop handler, the HTML events work fine except panics because the drop handler was expected to have set the most recent file: https://github.com/DioxusLabs/dioxus/blob/9ffd4b816b8ce291b6ca66cb5c871e107a964f6b/packages/desktop/src/webview.rs#L99

Most likely related: https://github.com/tauri-apps/wry/issues/904

I'm looking into creating a solution.

**Comment by Boscop** [comment]
Ah..
Thanks for looking into a solution for this :)

**Comment by rubcc95** [comment]
Any sotulion for this? Still panicking on drop file on Windows 10 when drag_drop_handler is disabled

**Comment by codenoobforreal** [comment]
Any update?
